tensorflow/examples/udacity/1_notmnist.ipynb

```diff
@@ -283,83 +283,140 @@
-        "  label_index = 0\n",
-        "  image_index = 0\n",
-        "  for folder in data_folders:\n",
-        "    print(folder)\n",
+        "def load_letter(folder, min_num_images):\n",
+        "    image_files = os.listdir(folder)\n",
+        "    dataset = np.ndarray(shape=(len(image_files), image_size, image_size), dtype=np.float32)\n",
+        "    image_index = 0\n",
+        "    print folder\n",
         "    for image in os.listdir(folder):\n",
-        "      if image_index >= max_num_images:\n",
-        "        raise Exception('More images than expected: %d >= %d' % (\n",
-        "          image_index, max_num_images))\n",
         "      image_file = os.path.join(folder, image)\n",
         "      try:\n",
-        "        image_data = (ndimage.imread(image_file).astype(float) -\n",
-        "                      pixel_depth / 2) / pixel_depth\n",
+        "        image_data = (ndimage.imread(image_file).astype(float) - pixel_depth / 2) / pixel_depth\n",
         "        if image_data.shape != (image_size, image_size):\n",
         "          raise Exception('Unexpected image shape: %s' % str(image_data.shape))\n",
         "        dataset[image_index, :, :] = image_data\n",
-        "        labels[image_index] = label_index\n",
         "        image_index += 1\n",
         "      except IOError as e:\n",
         "        print('Could not read:', image_file, ':', e, '- it\\'s ok, skipping.')\n",
-        "    label_index += 1\n",
-        "  num_images = image_index\n",
-        "  dataset = dataset[0:num_images, :, :]\n",
-        "  labels = labels[0:num_images]\n",
-        "  if num_images < min_num_images:\n",
-        "    raise Exception('Many fewer images than expected: %d < %d' % (\n",
-        "        num_images, min_num_images))\n",
-        "  print('Full dataset tensor:', dataset.shape)\n",
-        "  print('Mean:', np.mean(dataset))\n",
-        "  print('Standard deviation:', np.std(dataset))\n",
-        "  print('Labels:', labels.shape)\n",
-        "  return dataset, labels\n",
-        "train_dataset, train_labels = load(train_folders, 450000, 550000)\n",
-        "test_dataset, test_labels = load(test_folders, 18000, 20000)"
+        "    \n",
+        "    num_images = image_index\n",
+        "    dataset = dataset[0:num_images, :, :]\n",
+        "    if num_images < min_num_images:\n",
+        "        raise Exception('Many fewer images than expected: %d < %d' % (num_images, min_num_images))\n",
+        "    \n",
+        "    print('Full dataset tensor:', dataset.shape)\n",
+        "    print('Mean:', np.mean(dataset))\n",
+        "    print('Standard deviation:', np.std(dataset))\n",
+        "    return dataset\n",
+        "        \n",
+        "def load(data_folders, min_num_images, num_classes):\n",
+        "  dataset_names = []\n",
+        "  for folder in data_folders:\n",
+        "    dataset = load_letter(folder, min_num_images / num_classes)\n",
```

Comment:
For clarity, I would slightly prefer load() to be passed a min_num_images_per_class, to be forwarded directly to load_letter()